src/components/pages/categories.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <div class="category-page">
+    <app-category-post
+      :access="access"
+      :error-message="errorMessage"
+      :done-message="doneMessage"
+      class="category-page__post"
+    />
+    <div class="category-page__list">
+      <app-category-list
+        :theads="theads"
+        :access="access"
+        :categories="categoryList"
+      />
+    </div>>
+  </div>
+</template>
+
+<script>
+import { CategoryList, CategoryPost } from '@Components/molecules';
+
+export default {
+  components: {
+    appCategoryList: CategoryList,
+    appCategoryPost: CategoryPost,
+  },
+  data() {
+    return {
+      theads: ['カテゴリー名'],
+    };
+  },
+  computed: {
+    access() {
+      return this.$store.getters['auth/access'];
+    },
+    categoryList() {
+      return this.$store.state.categories.categoryList;
+    },
+    errorMessage() {
+      return this.$store.state.categories.errorMessage;
+    },
+    doneMessage() {
+      return this.$store.state.categories.doneMessage;
+    },
+  },
+  created() {
+    this.fetchCategories();
+  },
+  methods: {
+    fetchCategories() {
+      if (this.$route.query.category) {
+        const { category } = this.$route.query;
+        this.title = category;
+        this.$store.dispatch('categories/filteredCategories', category)
+          .then(() => {
+            if (this.$store.state.categories.categoryList.length === 0) {
+              this.$router.push({ path: '/notfound' });
+            }
+          }).catch(() => {
+            // console.log(err);
+          });
+      } else {
+        this.$store.dispatch('categories/getAllCategories');
+      }
```

Comment:
[must]

今回、`if (this.$route.query.category)` で条件を設定する必要はないかなと思います。
実際、画面設計書にもこのようなケースがあるというような記載はなかったかなと思います。
なので、以下の記述のみで良いかと思います。

```suggestion
        this.$store.dispatch('categories/getAllCategories');
```